modules/apps/foundation/frontend-theme/responsive-boat-theme/src/js/main.js

```diff
@@ -0,0 +1,44 @@
+
+// This function gets loaded when everything, including the portlets, is on the page
+
+Liferay.on('allPortletsReady', myPageJS);
+
+// Function Containing All of My Event Handlers
+
+function myPageJS() {
+	let nav = $('#navigation');
+	let navBtnDiv = $('#navigation div.nav-btn');
+	let portletParagraphs = $('#content .portlet-body p');
+
+	// SCROLL EVENT HANDLER
+
+	document.onscroll = function() {
+
+		// Toggle 'scrolled' class on nav based on scrolled location
+
+		if (window.scrollY >= 140)
+			nav.hasClass('scrolled') ? '' : nav.addClass('scrolled');
+		else if (nav.hasClass('scrolled'))
+			nav.removeClass('scrolled');
+
+		// Toggle 'visible' class to make portlet content visible when scrolled to
+
+		for (let i = 0; i < portletParagraphs.length; i++) {
+			if (!portletParagraphs.eq(i).hasClass('visible') && portletParagraphs[i].getBoundingClientRect().top - window.innerHeight <= 0)
+				portletParagraphs.eq(i).addClass('visible');
+		}
+	};
+
+	// CLICK EVENT HANDLER
+
+	navBtnDiv[0].onclick = function() {
+		let portletDiv = $('#navigation .portlet');
+
+		// Change maximum height of nav on click to reveal or hide nav links
+		
+		nav.toggleClass('expand');
+		portletDiv.slideToggle(700, function() {
+			portletDiv.css('display', portletDiv.css('display').replace('none',''));
+		});
```

Comment:
We should format this JS block like:
```
portletDiv.slideToggle(
    700,
    function() {
        portletDiv.css('display', portletDiv.css('display').replace('none',''));
    }
);
```

Reply:
It looks like you are trying to set the display value of this and replace it if it is set to none.  Why not just show the node instead of toggling the value like this?

Reply:
I am removing the inline-styling because it's taking precedence over my _custom.scss rules. When the inline display is set to none, then my responsive css no longer works and the navigation portlet and links stay hidden when I resize the window back to full screen. This was more annoying than anything, and I could also just re-expand the navigation in mobile view so that it shows up again in full screen desktop view.

Reply:
If you were to use jQuery to show the node, wouldn't that do the same thing as what you are doing?

Reply:
I don't think so. This was my logic behind that code:

1. When window width is > 768px, nav portlet is visible
2. When window width is <= 768px, the _custom.scss media query rule (display: none;) will hide the nav portlet
3. When nav button is clicked, my jQuery will expand the nav portlet and add an inline rule (display: block;) to override my css and make the nav portlet visible again
4. When nav button is clicked again, my jQuery will collapse the nav portlet and change the inline rule (display: none;) which overrides my css. This makes the nav portlet hidden at all times no matter what the width of the window is, and the nav bar disappears completely when I full screen the window again.
5. After the nav portlet is finished collasping, an additional jQuery function will remove the inline rule and allow the _custom.css media query rule (display: none;) to take over. I can now resize my window and the nav bar will be visible in full screen.

Is there a way that I could optimize this process?

Reply:
Let me think about it and see when I deploy your theme after your submission tomorrow to give you ideas and feedback about this.